docs/topic4.md

```diff
@@ -1,3 +1,170 @@
-# Topic 4 - Frameworks
+# Topic 4 - Javascript - Design patterns
 
-[<- To index](../README.md#title) - [Topic 5 - Project ->](./topic5.md)
\ No newline at end of file
+In this Topic we will focus on learning how JavaScript approaches Object-Oriented programming and
+some patterns to group related functionality together.
+
+If you come from Java, or .NET you will find yourself a little bit lost at the beggining.
+ECMAScript6 provides a layer of syntactic sugar over the previous version (5.1) that is expected to simplify the language.
+But there aren't the classic classes from another Object-Oriented languages
+
+**Documentation**
+
+- [Object-oriented JS](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Objects/Object-oriented_JS)
+- [Object prototypes](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Objects/Object_prototypes)
+- [Inheritance](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Objects/Inheritance)
+- [Module Patterns](https://www.freecodecamp.org/news/javascript-modules-a-beginner-s-guide-783f7d7a5fcc/)
+- [ES6 Modules](https://ponyfoo.com/articles/es6-modules-in-depth)
+
+**Extra documentation**
+
+- Understand how [prototypes](http://yehudakatz.com/2011/08/12/understanding-prototypes-in-javascript/) works in ECMAScript 5.1
+
+- Read the Chapter on JavaScript classes from [*Understanding ECMAScript6*](https://leanpub.com/understandinges6/read#leanpub-auto-classes)
+
+
+**Practice**
+
+1.  Creating classes
+
+    *   Create a Movie Class with the following structure
+
+        <table>
```

Comment:
I think the templates could be moved inside assets as the other one, makes sense?

Reply:
Esto no es código HTML, se esta usando para estilar el Markdown